smart-contracts/contracts/cl-vault/src/vault/range.rs

```diff
@@ -134,10 +135,16 @@ pub fn handle_withdraw_position_reply(
 
     let modify_range_state = MODIFY_RANGE_STATE.load(deps.storage)?.unwrap();
     let pool_config = POOL_CONFIG.load(deps.storage)?;
-    // what about funds sent to the vault via banksend, what about airdrops/other ways this would not be the total deposited balance
-    // todo: Test that one-sided withdraw wouldn't error here (it shouldn't)
-    let amount0: Uint128 = msg.amount0.parse()?;
-    let amount1: Uint128 = msg.amount1.parse()?;
+
+    let mut amount0: Uint128 = msg.amount0.parse()?;
+    let mut amount1: Uint128 = msg.amount1.parse()?;
+
+    let unused_balances = get_unused_balances(deps.storage, &deps.querier, &env)?;
```

Comment:
This needs to be tested, can be done by adding free balance to `test_execute_update_range`